fearless/Common/Protocols/KeyboardAdoptable.swift

```diff
@@ -14,21 +14,16 @@ extension KeyboardAdoptable {
             return
         }
 
-        keyboardHandler = KeyboardHandler(with: nil)
-        keyboardHandler?.animateOnFrameChange = { [weak self] keyboardFrame in
-            self?.updateWhileKeyboardFrameChanging(keyboardFrame)
-        }
+        keyboardHandler = FearlessKeyboardHandler(with: nil)
     }
 
     func clearKeyboardHandler() {
         keyboardHandler = nil
     }
 }
 
-protocol KeyboardViewAdoptable: KeyboardAdoptable {
-    var target: UIView? { get }
-    var currentKeyboardFrame: CGRect? { get set }
-    var shouldApplyKeyboardFrame: Bool { get }
+protocol KeyboardViewAdoptable: KeyboardAdoptable, KeyboardHandlerDelegate {
+    var target: Constraint? { get }
```

Comment:
what if I use frame for view? I can use offsetFromKeyboardWithInset?

Reply:
And we can't open pull request in SoraFoundation, because independent from SnapKit

Reply:
KeyboardAdoptable is not part of SoraFoundation, only KeyboardHandler. Use constraint is more safety and obvious than uiview